src/workerd/jsg/jsg.h

```diff
@@ -2881,7 +2875,8 @@ class Lock {
 
   // Resolve a module namespace from the given specifier.
   // This variation includes modules from the worker bundle.
-  kj::Maybe<JsObject> resolveModule(kj::StringPtr specifier);
+  kj::Maybe<JsObject> resolveModule(
+      kj::StringPtr specifier, RequireEsm requireEsm = RequireEsm::NO);
```

Comment:
Reviewer Note: The `resolveModule(...)` method was added a while ago. It is a single simplified entry point for resolving the named module from whichever module registry implementation is being used. What it previously did not do, however, is differentiate between whether the module is an ESM module or not, and when we are wanting to grab the entrypoint module we need to restrict it to ESM modules for backwards compat, therefore we add the option.